qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -132,89 +133,335 @@ def circuits(self) -> List[QuantumCircuit]:
 
-        Returns:
-            A list of :class:`QuantumCircuit`.
         """
-        circuits = []
-        for length in lengths if self._full_sampling else [lengths[-1]]:
-            elements = self._clifford_utils.random_clifford_circuits(self.num_qubits, length, rng)
-            element_lengths = [len(elements)] if self._full_sampling else lengths
-            circuits += self._generate_circuit(elements, element_lengths)
-        return circuits
+        transpiled_circs_file = CliffordUtils.file_name(
+            self.num_qubits, self.transpile_options.basis_gates
+        )
+        n = self.num_qubits
+        if self._transpiled_cliff_circuits[n] is None:
+            if os.path.isfile(transpiled_circs_file):
+                with open(transpiled_circs_file, "rb") as fd:
+                    self._transpiled_cliff_circuits[n] = qpy.load(fd)
+            else:
+                raise QiskitError(
+                    f"File {transpiled_circs_file} does not exist. "
+                    "Use generate_transpile_circuits.py to generate this file"
+                )
 
-    def _generate_circuit(
-        self, elements: Iterable[Clifford], lengths: Iterable[int]
+    def _build_rb_circuits(
+        self, lengths: List[int], rng: Generator, interleaved_element: QuantumCircuit = None
     ) -> List[QuantumCircuit]:
-        """Return the RB circuits constructed from the given element list.
-
+        """
+        build_rb_circuits
         Args:
-            elements: A list of Clifford elements
-            lengths: A list of RB sequences lengths.
+                lengths: A list of RB sequence lengths. We create random circuits
+                         where the number of cliffords in each is defined in lengths.
+                rng: Generator object for random number generation.
+                     If None, default_rng will be used.
+                interleaved_element: the interleaved element as a QuantumCircuit, if it exists.
 
         Returns:
-            A list of :class:`QuantumCircuit`s.
+                The transpiled RB circuits.
 
         Additional information:
-            The circuits are constructed iteratively; each circuit is obtained
-            by extending the previous circuit (without the inversion and measurement gates)
+            To create the RB circuit, we use a mapping between Cliffords and integers
+            defined in the file clifford_data.py. The operations compose and inverse are much faster
+            when performed on the integers rather than on the Cliffords themselves.
         """
-        qubits = list(range(self.num_qubits))
-        circuits = []
+        if self._full_sampling:
+            return self._build_rb_circuits_full_sampling(lengths, rng, interleaved_element)
+        is_interleaved = interleaved_element is not None
+        max_qubit = max(self.physical_qubits) + 1
+        all_rb_circuits = []
+
+        if is_interleaved:
```

Comment:
This breaks the current structure where `StandardRB` did not know about `InterleavedRB`. Can't say if it's good or bad, but I usually prefer ensuring classes know as little about each other as possible (so if someone wants to understand interleaved RB, all the relevant interleaved logic will be in the interleaved rb file).

Reply:
I agree with this comment. I also was not sure if this was best. The other option would be to copy the `_build` methods to `InterleavedRB`. There would be a lot of code duplication, because I didn't see any reasonable way to break these down into smaller methods.
But if you think this is better, I will make this change.